### DISCUSSION TITLE
Browser version

### DISCUSSION THREAD

**Discussion by The24thDS** [umami-software/umami#1385] [Ideas]
I think one of the most important information with analytics is the browser versions that your users are using.

We all know how older versions don't support all the features of the newer ones. It would be useful for a team of developers to know what browser versions their target audience is using.

Can this be added to the enhancements list?
Let me know if I can help with it, I'm willing to take a look at the code :)

**Comment by kisexu** [comment]
An important requirement indeed. It can help the project to determine which versions of browsers to adapt.

**Comment by tua-Mascot** [comment]
It would also be nice to add OS versions (especially for iOS, macOS and Android) 🙏

**Comment by tua-Mascot** [comment]
@mikecao could you please give us some answer for this idea request?

**Comment by arjoonn-s** [comment]
It would actually be very useful to have all the versions mentioned in the user agent somewhere. For example having the same browser version on a different OS version can also introduce subtle bugs in the rendering of the website for example.

**Comment by benjaminrancourt** [comment]
A real example of the value added by the browser version is that without it, I am unable to drop support for legacy clients, such for [SSL configuration](https://ssl-config.mozilla.org/#server=nginx&version=1.17.7&config=modern&openssl=1.1.1k&guideline=5.7). Do I have users who are still using browser versions lower than these (Firefox 63, Android 10.0, Chrome 70, Edge 75, Java 11, OpenSSL 1.1.1, Opera 57 and Safari 12.1)?

So for now I'm stuck with the "Intermediate" configuration because I don't want to provide a broken experience by switching without any knowledge to the "Modern" configuration...

![image](https://github.com/umami-software/umami/assets/7907713/d17cf534-50c5-497c-90c0-dbf3d0a3f67c)
